### PR TITLE
Optimisation of memory usage. Reduced allocations of objects.

### PR DESCRIPTION
**Pull request by AlexanderMy**

### CODE REVIEW

**Comment by niemyjski** [comment]
@AlexanderMy Thanks for submitting the PR, can you share any benchmarks vs the baseline you ran against. We have some benchmarks setup in a benchmark project in this repo.

File: src/TinyIoC/TinyIoC.cs [hunk 1]
```diff
@@ -3580,53 +3580,54 @@ where localType.IsAssignableFrom(implementationType)
...
 #endif
-                t => (t.GetConstructors(BindingFlags.Instance | BindingFlags.Public).Length == 0) && !(t.IsInterface() || t.IsAbstract()),
-            };
+            t => (t.GetConstructors(BindingFlags.Instance | BindingFlags.Public).Length == 0) && !(t.IsInterface() || t.IsAbstract()),
+        };
 
-            if (registrationPredicate != null)
+        private bool IsIgnoredType(Type type, Func<Type, bool> registrationPredicate)
+        {
+            if (registrationPredicate != null && !registrationPredicate(type))
```
Comment (niemyjski) [inline src/TinyIoC/TinyIoC.cs:3622]: Seems like this should be stored in a temp variable and added to the ignore checks below (saved an extra lookup)

Comment (AlexanderMy) [reply src/TinyIoC/TinyIoC.cs:3622]: This is done as you said or am I missing something?

**Comment by AlexanderMy** [comment]
> @AlexanderMy Thanks for submitting the PR, can you share any benchmarks vs the baseline you ran against. We have some benchmarks setup in a benchmark project in this repo.

Please find my results:
Screenshot 2021-06-22 at 17 25 01

**Comment by niemyjski** [comment]
Thanks for the PR! Let me know if you need a release out there soon?

**Comment by AlexanderMy** [comment]
> Thanks for the PR! Let me know if you need a release out there soon?

Thanks for merging. We don't need the release just now, thanks.

**Pull request merged**